sweepai/api.py

```diff
@@ -389,3 +393,34 @@ async def webhook(raw_request: Request):
         logger.warning(f"Failed to parse request: {e}")
         raise HTTPException(status_code=422, detail="Failed to parse request")
     return {"success": True}
+
+@stub.function(**FUNCTION_SETTINGS)
+def update_sweep_prs(
+    repo_full_name: str,
+    installation_id: int
+):
+    # Get a Github client
+    g = get_github_client(installation_id)
+    
+    # Get the repository
+    repo = g.get_repo(repo_full_name)
+    
+    # Get all open pull requests created by Sweep
+    pulls = repo.get_pulls(state='open', head='sweep')
+    
+    # For each pull request, attempt to merge the changes from the default branch into the pull request branch
+    for pr in pulls:
+        try:
+            # Create a new branch from the head branch of the pull request
+            new_branch = repo.create_git_ref(ref=f"refs/heads/sweep-merge-{pr.number}", sha=pr.head.sha)
+            
+            # Merge the base branch into the new branch
+            base = repo.get_git_ref(f"heads/{pr.base.ref}")
+            repo.merge(new_branch.ref, base.object.sha)
+            
+            # Create a new pull request with the new branch as the head and the original head branch of the pull request as the base
+            repo.create_pull(title=f"Merge changes from default branch into PR #{pr.number}", body="", head=new_branch.ref, base=pr.head.ref)
```

Comment:
instead of making a PR just merge directly from default branch into feature branch